src/main/java/uk/gov/hmcts/reform/hmc/helper/hmi/HmiHearingResponseMapper.java

```diff
@@ -78,6 +85,24 @@ public HearingEntity mapHmiHearingToEntity(HearingResponse hearing, HearingEntit
         return hearingEntity;
     }
 
+    private List<HearingSession> findUniqueHearingSessionsPerDay(List<HearingSession> hearingSessions) {
+        MultiValuedMap<LocalDateTime, HearingSession> uniqueDays = new ArrayListValuedHashMap<>();
+
```

Comment:
We could make everything to use independent streams to resolve everything. Instead of passing an MultiValuedMap throught loops.
````
//Map <LocalDateTime,List<LocalDateTime,HearingSession>>
val map2 = hearingSessions.stream()
  .collect(Collectors.groupingBy(hearingSession -> hearingSession.getHearingStartTime().truncatedTo(ChronoUnit.DAYS)));

map2.keySet().stream().map(currentEntry -> {
        val date currentEntry.geKey();
        HearingSession hearingSessionWithEarliestStartTime = currentEntry.geKey()
        HearingSession hearingSessionWithLatestEndTime =
        
        hearingSessionWithEarliestStartTime.setHearingEndTime(hearingSessionWithLatestEndTime.getHearingEndTime());
        return hearingSessionWithEarliestStartTime;
}
`````
But It will imples to re write it and it would not add a huge value.  Not needed